Setup.bat

```diff
@@ -6,6 +6,15 @@ pushd "%~dp0"
 
 call :MarkStartOfBlock "%~0"
 
+set ProjectDirectory=%1

```

Comment:
I'd add quotes around all variables and print messages that refer to paths.
Has this been tested in a folder structure with spaces in it?

Reply:
Adjusted all variables involving a path to be declared with quotes. This makes all usages not require quotes.

Reply:
Also tested and working with spaces in paths :D